src/border_agent/border_agent.cpp

```diff
@@ -378,6 +378,19 @@ void BorderAgent::PublishMeshCopService(void)
     // "xa" stands for Extended MAC Address (64-bit) of the Thread Interface of the Border Agent.
     txtList.emplace_back("xa", extAddr->m8, sizeof(extAddr->m8));
 
+    // An unique ID is required for adding the Thread network credentials of this BR to Android
+    // or iOS system. Products can use the dbus "UpdateVendorMeshCopTxtEntries" method to set a
+    // vendor-specific ID, or otherwise, enable the feature here set ID to Extended MAC Address
+    // by padding zeros at the tail.
+#if OTBR_ENABLE_OTBR_PUBLISH_ID_AS_EXT_ADDRESS
+    {
+        uint8_t id[16] = {0};
+
+        memcpy(id, extAddr->m8, sizeof(extAddr->m8));
```

Comment:
is it better to put into the last 8 bytes? It might be easier for us to represent as `0x....`.

Reply:
Agree. But unfortunately there is legacy code in Android/GMS that zeros are padding at the tail, so here to follow it to avoid potential compatibility issues.